improver_tests/synthetic_data/test_generate_metadata.py

```diff
@@ -0,0 +1,556 @@
+# CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE)
+# ARISING IN ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE
+# POSSIBILITY OF SUCH DAMAGE.
+"""Tests for metadata cube generation."""
+
+from datetime import datetime
+
+import iris
+import numpy as np
+import pytest
+
+from improver.grids import GLOBAL_GRID_CCRS, STANDARD_GRID_CCRS
+from improver.synthetic_data.generate_metadata import generate_metadata
+from improver.utilities.temporal import datetime_to_iris_time, iris_time_to_datetime
+
+NAME_DEFAULT = "air_pressure_at_sea_level"
+UNITS_DEFAULT = "Pa"
+SPATIAL_GRID_DEFAULT = "latlon"
+NPOINTS_DEFAULT = 71
+ENSEMBLE_MEMBERS_DEFAULT = 8
+TIME_DEFAULT = datetime(2017, 11, 10, 4, 0)
+FRT_DEFAULT = datetime(2017, 11, 10, 0, 0)
+FORECAST_PERIOD_DEFAULT = 14400
+NDIMS_DEFAULT = 3
+ATTRIBUTES_DEFAULT = {}
+RELATIVE_TO_THRESHOLD_DEFAULT = "above"
+SPATIAL_GRID_ATTRIBUTE_DEFAULTS = {
+    "latlon": {
+        "y": "latitude",
+        "x": "longitude",
+        "grid_spacing": 0.02,
+        "units": "degrees",
+        "coord_system": GLOBAL_GRID_CCRS,
+    },
+    "equalarea": {
+        "y": "projection_y_coordinate",
+        "x": "projection_x_coordinate",
+        "grid_spacing": 2000,
+        "units": "metres",
+        "coord_system": STANDARD_GRID_CCRS,
+    },
+}
+
+
+def _check_cube_shape_different(cube):
+    """ Asserts that cube shape has been changed from default but name, units and
+    attributes are unchanged """
+    default_cube = generate_metadata()
+    assert cube.shape != default_cube.shape
+    assert iris.util.describe_diff(cube, default_cube) is None
+
+
+def test_default():
+    """ Tests default metadata cube generated """
+    cube = generate_metadata()
+
+    assert cube.name() == NAME_DEFAULT
+    assert cube.standard_name == NAME_DEFAULT
+    assert cube.units == UNITS_DEFAULT
+
+    assert cube.ndim == NDIMS_DEFAULT
+    assert cube.shape == (ENSEMBLE_MEMBERS_DEFAULT, NPOINTS_DEFAULT, NPOINTS_DEFAULT)
+
+    spatial_grid_values = SPATIAL_GRID_ATTRIBUTE_DEFAULTS[SPATIAL_GRID_DEFAULT]
+    assert cube.coords()[0].name() == "realization"
+    assert cube.coords()[1].name() == spatial_grid_values["y"]
+    assert cube.coords()[2].name() == spatial_grid_values["x"]
+
+    for axis in ("y", "x"):
+        assert cube.coord(axis=axis).units == spatial_grid_values["units"]
+        assert cube.coord(axis=axis).coord_system == spatial_grid_values["coord_system"]
+        assert np.diff(cube.coord(axis=axis).points)[0] == pytest.approx(
+            spatial_grid_values["grid_spacing"]
+        )
+
+    assert np.count_nonzero(cube.data) == 0
+
+    assert iris_time_to_datetime(cube.coord("time"))[0] == TIME_DEFAULT
+    assert cube.coord("time").bounds is None
+    assert (
+        iris_time_to_datetime(cube.coord("forecast_reference_time"))[0] == FRT_DEFAULT
+    )
+    print(cube.coord("forecast_period").points)
```

Comment:
Print statement to be removed.

Reply:
Removed [here](https://github.com/metoppv/improver/pull/1326/commits/a8ae93b36fa2d7574acb2ab8f36b20f272d2a85f#diff-dfb229535f6ef5488fe1cf6ea49f02aeL110).